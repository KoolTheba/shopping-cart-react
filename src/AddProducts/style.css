.products-list {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    flex-wrap: wrap;
    margin-bottom: 5rem;
}

.product-image img {
    width: 150px;
    height: 150px;
}

.product-info {
    text-align: center;
}

p {
    margin-bottom: 15px;
}

.add-to-cart-button {
    cursor: pointer;
    font-weight: bold;
    width: 145px;
    height: 2.4rem;
    font-size: 1rem;
}

.add-to-cart-button:hover {
    background-color: #799F0C;
}

.cart-icon {
    margin-left: 0.5rem;
}