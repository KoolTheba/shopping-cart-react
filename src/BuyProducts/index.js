import React from 'react'
import PropTypes from 'prop-types'

import { makeStyles } from '@material-ui/core/styles'
import Table from '@material-ui/core/Table'
import TableBody from '@material-ui/core/TableBody'
import TableContainer from '@material-ui/core/TableContainer'
import TableHead from '@material-ui/core/TableHead'
import TableRow from '@material-ui/core/TableRow'
import TableCell from '@material-ui/core/TableCell'

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faShoppingBasket } from '@fortawesome/free-solid-svg-icons'

import BuyProductItem from '../BuyProductItem'
import CleanCart from '../CleanCart'

const useStyles = makeStyles({
    buyProducts: {
      width: '100%',
      margin: '125px 0 50px',
    },
    table: {
      minWidth: 650,
      background: 'linear-gradient(45deg, #799F0C 30%, #799F0C 90%)',
      border: 0,
      borderRadius: 3,
      boxShadow: '0 3px 5px 2px rgba(255, 105, 135, .3)',
      color: 'white',
      height: 48,
      padding: '0 30px',
    },
    tableHead: {
        color: '#292C34',
        fontSize: 20,
        fontWeight: "bold",
        fontFamily: 'sans-serif',
    }
});

function BuyProducts (props) {
    const classes = useStyles()

    const productsList = Object.keys(props.cartList).map(id => {
        const {image, name, price} = props.products[id];
        return {
            id,
            quantity: props.cartList[id],
            image, 
            name, 
            price
        }
    })

    return(
        <>
        <section className={classes.buyProducts} id='products-list-section'>
            <h3>Your shopping bag <FontAwesomeIcon icon={faShoppingBasket}/></h3>
            <TableContainer>
                <Table className={classes.table} aria-label="simple table">
                    <TableHead>
                        <TableRow>
                            <TableCell className={classes.tableHead} align="center">Product</TableCell>
                            <TableCell className={classes.tableHead} align='left'></TableCell>
                            <TableCell className={classes.tableHead} align="center">Remove</TableCell>
                            <TableCell className={classes.tableHead} align="center">Unit price</TableCell>
                            <TableCell className={classes.tableHead} align="center">Quantity</TableCell>
                            <TableCell className={classes.tableHead} align="center">Total</TableCell>
                        </TableRow>
                    </TableHead>
                    <TableBody>
                        {productsList.length > 0 &&
                            productsList.map(product => (
                                <BuyProductItem
                                    key={product.id}
                                    product={product}
                                    addUnit={props.addUnit}
                                    removeUnit={props.removeUnit}
                                    removeListItem={props.removeListItem}
                                />
                            ))
                        }
                    </TableBody>
                </Table>
            </TableContainer>
            <CleanCart 
                cleanShoppingList={props.cleanShoppingList}
            />
        </section>
        </>
    )
}

BuyProducts.propTypes = {
    cartList: PropTypes.object,
    products: PropTypes.object,
    addUnit: PropTypes.func,
    removeUnit: PropTypes.func,
    removeListItem: PropTypes.func,
    cleanShoppingList: PropTypes.func,
}

export default BuyProducts