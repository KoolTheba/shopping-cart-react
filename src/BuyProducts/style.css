.products-to-buy {
    width: 100%;
    border: 1px solid white;
}

.table-head {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
}