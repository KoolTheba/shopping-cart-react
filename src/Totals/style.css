.totals-section {
    text-align: end;
    margin-top: 50px;
}