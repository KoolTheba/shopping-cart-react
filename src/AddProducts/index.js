import React from 'react'
import './style.css'
import data from '../data/products.json'
console.log(data)

export default class AddProducts extends React.Component {

    state = {
        products: []
    }

    componentDidMount(){
        this.setState({ products: data })
    }

    handleAddProduct = (id) => {
        console.log('ID-----', id)
    }

    render(){
        const { products } = this.state

        return(
            <>
            <h3>Cool products to add to your shopping list</h3>
            <section className='products-list'>
            {Object.entries(products).map(([id, info]) => (
                <div 
                    className="product-card"
                    id={id}
                    key={id}
                >
                    <div className="product-image">
                        <img src={info.image} alt={info.image}/>
                    </div>
                    <div className="product-info">
                        <p>{info.name}</p>
                        <p>â‚¬{info.price}</p>
                        <button
                            onClick={() => this.handleAddProduct(id)}
                        >Add to Cart</button>
                    </div>
                </div>
            ))}
            </section>
            </>
        )
    }
}