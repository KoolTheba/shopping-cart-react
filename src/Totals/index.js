import React from 'react'
import './style.css'

function Totals (){
    return(
        <>
        <h5>Total: <span>XXXX €</span></h5>
        <h5>Discount Code: 
            <input placeholder='enter discount code...'></input>
            <button>Validate Code</button>
        </h5>
        <h5>Total Cart: <span>XXXX €</span></h5>
        </>
    )
}

export default Totals