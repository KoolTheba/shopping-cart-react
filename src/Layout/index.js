import React from 'react'
import './style.css'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faShoppingCart} from '@fortawesome/free-solid-svg-icons'


import AddProducts from '../AddProducts'
import BuyProducts from '../BuyProducts'
import Totals from '../Totals'

function Layout() {
  return (
    <div>
      <header className="App-header">
        <h1>Cart <FontAwesomeIcon icon={faShoppingCart}/></h1>
      </header>
      <AddProducts />
      <BuyProducts />
      <Totals />
    </div>
  );
}

export default Layout;
