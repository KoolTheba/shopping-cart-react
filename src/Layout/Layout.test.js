import React from 'react'
import Layout from './'
import { shallow, mount } from 'enzyme'
import AddProducts from '../AddProducts'
import BuyProducts from '../BuyProducts'
import Totals from '../Totals'

describe('Layout component', () => {
    it('renders loading when state.loading is true', () => {
        const wrapper = mount(<Layout />)
        wrapper.setState({ loading: true })
        expect(wrapper.contains(<h1>loading...</h1>)).toBe(true)
        expect(wrapper.exists(AddProducts)).toBe(false)
    });

    it('does not render loading when state.loading is true', () => {
        const wrapper = mount(<Layout />)
        wrapper.setState({ loading: false });
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.exists(AddProducts)).toBe(true)
        expect(wrapper.exists(BuyProducts)).toBe(true)
        expect(wrapper.exists(Totals)).toBe(true)
    });

    it('all Layout methods should be typeof functions', () => {
        const wrapper = shallow(<Layout />)
        expect(typeof wrapper.instance().handleAddProduct).toBe('function')
        expect(typeof wrapper.instance().handleAddProduct).toBe('function')
        expect(typeof wrapper.instance().handleAddQuantity).toBe('function')
        expect(typeof wrapper.instance().handleRemoveQuantity).toBe('function')
        expect(typeof wrapper.instance().handleRemoveListItem).toBe('function')
        expect(typeof wrapper.instance().handleCleanShoppingList).toBe('function')
        expect(typeof wrapper.instance().handleInputChange).toBe('function')
        expect(typeof wrapper.instance().handleValidateCode).toBe('function')
    });
});
