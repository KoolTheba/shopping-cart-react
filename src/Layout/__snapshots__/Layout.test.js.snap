// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component does not render loading when state.loading is true 1`] = `ReactWrapper {}`;
